micropub.php

```diff
@@ -810,31 +810,44 @@ private static function check_error( $result ) {
 			return $result;
 		}
 
+
+		public static function get_micropub_endpoint() {
+			return apply_filters( 'micropub_endpoint', site_url( '?micropub=endpoint' ) );
+		}
+
+		public static function get_authentication_endpoint() {
+			return apply_filters( 'micropub_authentication_endpoint', MICROPUB_AUTHENTICATION_ENDPOINT );
+		}
+
+		public static function get_token_endpoint() {
+			return apply_filters( 'micropub_endpoint', MICROPUB_TOKEN_ENDPOINT );
+		}
+
 		/**
 	 * The micropub autodicovery meta tags
 	 */
 		public static function html_header() {
-			echo '<link rel="micropub" href="' . site_url( '?micropub=endpoint' ) . '">';
-			echo '<link rel="authorization_endpoint" href="' . MICROPUB_AUTHENTICATION_ENDPOINT . '">';
-			echo '<link rel="token_endpoint" href="' . MICROPUB_TOKEN_ENDPOINT . '">';
+			printf( '<link rel="micropub" href="%s" />' . PHP_EOL, static::get_micropub_endpoint() );
```

Comment:
oh huh, i didn't realize the closing `/` was allowed (optional) in empty elements: https://www.w3.org/TR/html/syntax.html#start-tags

i'm fine either way then.